vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java

```diff
@@ -668,6 +664,49 @@ public MongoClient dropIndex(String collection, String indexName, Handler<AsyncR
     return this;
   }
 
+  @Override
+  public MongoClient watch(String collection, JsonArray pipeline, WatchOptions options, Handler<AsyncResult<MongoClientWatchIterable>> resultHandler) {
+    requireNonNull(collection);
+    requireNonNull(pipeline);
+
+    if (options == null) {
+      options = new WatchOptions();
+    } else {
+      if (options.getFullDocument() != null
+        && !options.getFullDocument().equals(WatchOptions.FULL_DOCUMENT_UPDATE_LOOKUP)) {
+        resultHandler.handle(Future.failedFuture(new IllegalArgumentException("fullDocument must be null or \"updatedLookup\"")));
+        return this;
+      }
+    }
+
+    try {
+      List<Bson> pipelineBson = new ArrayList<>();
+      for (int i = 0; i < pipeline.size(); i++) {
+        pipelineBson.add(toBson(pipeline.getJsonObject(i)));
+      }
+
+
+      ChangeStreamIterable<JsonObject> view = getCollection(collection)
+        .watch(pipelineBson)
+        .fullDocument((options.getFullDocument() != null && options.getFullDocument().equals(WatchOptions.FULL_DOCUMENT_UPDATE_LOOKUP)) ? FullDocument.UPDATE_LOOKUP : FullDocument.DEFAULT)
+        .batchSize(options.getBatchSize());
+      if (options.getResumeAfter() != null) {
+        view.resumeAfter((BsonDocument) toBson(options.getResumeAfter()));
+      }
+      if (options.getMaxAwaitTimeMS() != WatchOptions.DEFAULT_MAX_AWAIT_TIME_MS) {
+        view.maxAwaitTime(options.getMaxAwaitTimeMS(), TimeUnit.MILLISECONDS);
+      }
+
+      Context context = vertx.getOrCreateContext();
+      MongoIterableStream<ChangeStreamDocument<JsonObject>> adaptor = new MongoIterableStream<>(view, context, options.getBatchSize());
+      resultHandler.handle(Future.succeededFuture(new MongoClientWatchIterableImpl(adaptor)));
+
+    } catch (Exception unhandledEx) {
```

Comment:
Are there not CheckedExceptions that can be caught?  Or are you worried about the random RuntimeExcpetion like an NPE?